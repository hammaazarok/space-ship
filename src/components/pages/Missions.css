.missions-table {
  margin: 0;
  text-align: start;
}

.desc {
    width: 60%;
}

td {
    vertical-align: middle
}
