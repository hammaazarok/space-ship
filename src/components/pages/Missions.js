import './Missions.css';
import React, { useEffect } from 'react';
import Table from 'react-bootstrap/Table';
import Button from 'react-bootstrap/Button';
import Badge from 'react-bootstrap/Badge';
import { useDispatch, useSelector } from 'react-redux';
import { addremoveMembertoMission, LoadMessions } from '../../redux/missions/missions';

const Missions = () => {
  const fetchMissions = async () => {
    const missions = [];
    await fetch('https://api.spacexdata.com/v3/missions')
      .then((res) => res.json())
      .then((json) => {
        json.forEach((e) => {
          const mission = {
            mission_id: e.mission_id,
            mission_name: e.mission_name,
            description: e.description,
            isMember: false,
          };
          missions.push(mission);
        });
      });
    return missions;
  };
  const dispatch = useDispatch();
  useEffect(() => { fetchMissions().then((res) => dispatch(LoadMessions(res))); }, []);
  const { error, loading, missions } = useSelector((state) => state.missions);
  const Missions = missions;

  console.log(error, loading, Missions);
  const toggleMember = (id) => {
    console.log(id);
    dispatch(addremoveMembertoMission(id));
  };
  return (
    <Table striped bordered hover size="sm" className="missions-table ml-5">
      <thead>
        <tr>
          <th>Mission</th>
          <th className="desc">Description</th>
          <th>Status</th>
          <th> </th>
        </tr>
      </thead>
      <tbody>
        { Missions.map((m) => (
          <tr key={m.mission_id}>
            <td><strong>{m.mission_name}</strong></td>
            <td>{m.description}</td>
            <td className="td-button">
              <h5>
                {m.isMember && (
                <Badge bg="info">Active Member</Badge>
                )}
                {!m.isMember && (
                <Badge bg="secondary">NOT A MEMBER</Badge>
                )}

              </h5>
            </td>
            <td className="td-button">
              {m.isMember && (
              <Button variant="outline-danger" onClick={() => toggleMember(m.mission_id)}>Leave Mission</Button>
              )}
              {!m.isMember && (
                <Button variant="outline-secondary" onClick={() => toggleMember(m.mission_id)}>Join Mission</Button>
              )}
            </td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

export default Missions;
