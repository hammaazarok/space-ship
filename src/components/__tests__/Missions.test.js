import React from 'react';
import Renderer from 'react-test-renderer';
import Missions from '../pages/Missions';

it('renders correctly', () => {
  const tree = Renderer
    .create(<Missions />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
