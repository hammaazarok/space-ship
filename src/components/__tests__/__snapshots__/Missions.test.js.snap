// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="table-container"
>
  <table
    className="missions-table ml-5 table-responsive table table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th
          className="desc"
        >
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
